astroquery/hsa/core.py

```diff
@@ -0,0 +1,364 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import cgi
+import os
+import re
+import shutil
+from pathlib import Path
+
+from astroquery import log
+from astroquery.exceptions import LoginError, InputWarning
+from astroquery.query import BaseQuery
+from astroquery.utils.tap.core import Tap
+
+from . import conf
+
+__all__ = ['HSA', 'HSAClass']
+
+
+class HSAClass(BaseQuery):
+
+    data_url = conf.DATA_ACTION
+    metadata_url = conf.METADATA_ACTION
+    timeout = conf.TIMEOUT
+
+    def __init__(self, tap_handler=None):
+        super(HSAClass, self).__init__()
+        if tap_handler is None:
+            self._tap = Tap(url=self.metadata_url)
+        else:
+            self._tap = tap_handler
+
+    def download_data(self, *, retrieval_type=None,
+                      observation_id=None,
+                      instrument_name=None,
+                      filename=None,
+                      verbose=False,
+                      cache=True,
+                      **kwargs):
```

Comment:
I've added the parameters in the docstring to the signature. I wouldn't add the rest of them because IMO there are a lot [see](http://archives.esac.esa.int/hsa/archive-help/cl/product_access/product_access.html#http_parameters)